src/indexer.rs

```diff
@@ -81,17 +89,59 @@ impl Indexer {
     }
 
     pub async fn start(&mut self) -> Result<()> {
-        handle_messages! {
-            on_bus self.bus,
-            listen<ConsensusEvent> cmd => {
-                match self.handle_consensus_event(cmd).await{
-                    Ok(_) => (),
-                    Err(e) => error!("Error while handling consensus event: {:#}", e),
+        if self.da_stream.is_none() {
+            handle_messages! {
+                on_bus self.bus,
+                listen<ConsensusEvent> cmd => {
+                    if let Err(e) = self.handle_consensus_event(cmd).await {
+                        error!("Error while handling consensus event: {:#}", e)
+                    }
+                }
+            }
+        } else {
+            handle_messages! {
+                    on_bus self.bus,
+                    cmd = self.da_stream.as_mut().expect("da_stream must exist").next() => {
+                    if let Some(Ok(cmd)) = cmd {
+                        let bytes = cmd;
+                        let block: Block = bincode::decode_from_slice(&bytes, bincode::config::standard()).unwrap().0;
+                        if let Err(e) = self.handle_block(block).await {
+                            error!("Error while handling block: {:#}", e);
+                        }
+                        SinkExt::<bytes::Bytes>::send(self.da_stream.as_mut().expect("da_stream must exist"), "ok".into()).await.unwrap();
```

Comment:
this "ok" sent over the wire : is it expected on the other side ? I can't find the place where it is read from the stream in data_availability. If not expected on the other side, maybe we can remove this line ? but maybe I missed something

Reply:
This is just the other end of the "ping" logic, which I could remove entirely. The actual message I'm sending is completely unimportant for now.

Reply:
ok got it